vendor/github.com/weaveworks/common/middleware/grpc_logging.go

```diff
@@ -17,6 +17,11 @@ const (
 	errorKey = "err"
 )
 
+// If an error implements Observe(), it will get called and GRPCServerLog will do nothing.
+type Observer interface {
```

Comment:
Can we document some of the expectations of implementers of this? For example, we expect if you implement `Observer` you also implement `error` and have a `.Unwrap()` method.

Reply:
> we expect if you implement Observer you also implement error and have a .Unwrap() method

I don't think we expect that, right? You'll have to implement `error` because this is checked on an error, but _we_ don't need that. And there's no need for the `Unwrap()` method.